### PR TITLE
EECS 281 Setup

### LINKED ISSUE #15: Tutorial consistency with other courses
It would be nice for students if the tutorials were consistent other courses, particularly EECS 281.  That would ease the transition into and out of EECS 280.

This is issue is *not* to imply that EECS 280 should change everything to match EECS 281, but I do want to make a laundry list of the differences.

### EECS 280 v. EECS 281
- Visual Studio and Xcode default project directory (see eecs280staff/tutorials#16)
- Xcode file input redirection hack
- Visual Studio getopt_long() hack
- Valgrind vs. address sanitizer
- Supported IDEs
- Makefile debug target by default in 280, optimized production target in 281.  EECS 281 has a universal Makefile.

### EECS 280 v. EECS 183
- Visual Studio and Xcode default project directory (see eecs280staff/tutorials#16)
- Supported IDEs

### LINKED ISSUE #36: EECS 281 Tutorial
It would be a nice service to publish an EECS 281 setup tutorial that uses the EECS 280 tutorials.

### PR DESCRIPTION
**Pull request by awdeorio**
Add an EECS 281 setup tutorial using the EECS 280 tutorials. [Preview](https://preview.sesh.rs/previews/eecs280staff/tutorials/6/setup_eecs281.html)

### Prerequisites
- [x] #42
- [x] #43
- [x] #44

### TODO
- [x] Major edits after #30 is merged
- [x] get opt starter code
- [x] get opt work around on Visual Studio
- [x] Configure visual debugger for CLI args and opts

### Related
Closes #15
Closes #36

### CODE REVIEW

**Comment by awdeorio** [comment]
@zjweiss I think this is ready for review!

File: docs/setup_eecs281.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+Download the EECS 281 Makefile.
+```console
+$ wget https://gitlab.umich.edu/eecs281/makefile/-/raw/main/Makefile
+$ make help
+EECS281 Advanced Makefile Help
+...
+```
+
+Edit these three lines in the Makefile.  Your values might be different, check the project spec for required file names.
```
Comment (zjweiss) [inline docs/setup_eecs281.md:48]: There are 4 things listed to be edited.

File: docs/setup_eecs281.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+# ...
+IDENTIFIER  = copy_me_from_project_spec
+# ...
+PROJECTFILE = main.cpp
+# ...
+EXECUTABLE = main
+```
+{: data-title="Makefile" }
+
+**Shortcut:** You can skip updating `PROJECTFILE` if your `main()` function is in one of these filenames: `main.cpp`, `project0.cpp`, `project1.cpp`, `project2.cpp`, `project3.cpp`, `project4.cpp`, `EXECUTABLE.cpp`.
```
Comment (zjweiss) [inline docs/setup_eecs281.md:60]: Add "or" between `project4.cpp` and `EXECUTABLE.cpp` to be grammatically correct.

**Review by zjweiss — APPROVED** [review]
LGTM. Just had 2 small nits.

**Comment by awdeorio** [comment]
@zjweiss Check my PR on the EECS 281 makefile repo: https://gitlab.umich.edu/eecs281/makefile/-/merge_requests/3

**Comment by awdeorio** [comment]
This is ready to go!

**Pull request merged**